packages/gatsby/src/schema/types/type-builders.ts

```diff
@@ -96,11 +106,20 @@ function buildScalarType<TSource, TContext>(
   }
 }
 
-function isGatsbyType(something: any): something is GatsbyGraphQLTypeKind {
+function isGatsbyType(
+  something: TypeOrTypeDef
+): something is GatsbyGraphQLType<any, any> {
   return (
     typeof something === `object` &&
+    !(something instanceof GraphQLScalarType) &&
+    !(something instanceof GraphQLObjectType) &&
+    !(something instanceof GraphQLInterfaceType) &&
+    !(something instanceof GraphQLUnionType) &&
+    !(something instanceof GraphQLEnumType) &&
+    !(something instanceof GraphQLList) &&
+    !(something instanceof GraphQLNonNull) &&
```

Comment:
This is required because [GraphQLScalarType](https://github.com/graphql/graphql-js/blob/v14.7.0/tstypes/type/definition.d.ts#L285), [GraphQLObjectType](https://github.com/graphql/graphql-js/blob/v14.7.0/tstypes/type/definition.d.ts#L371), [GraphQLInterfaceType](https://github.com/graphql/graphql-js/blob/v14.7.0/tstypes/type/definition.d.ts#L553), [GraphQLUnionType](https://github.com/graphql/graphql-js/blob/v14.7.0/tstypes/type/definition.d.ts#L619), [GraphQLEnumType](https://github.com/graphql/graphql-js/blob/v14.7.0/tstypes/type/definition.d.ts#L677), [GraphQLList](https://github.com/graphql/graphql-js/blob/v14.7.0/tstypes/type/definition.d.ts#L168) and [GraphQLNonNull](https://github.com/graphql/graphql-js/blob/v14.7.0/tstypes/type/definition.d.ts#L202) do not have `kind`